### PR TITLE
Product Creation AI: Use eligibility checker in coordinator

### LINKED ISSUE #10689: Eligibility check
Check `ai-assistant` feature availability from `me/sites` response.

Update Networking and Storage layers to read this info and add it to `Site`

More details 
p1694608029232419/1694598270.681329-slack-C03L1NF1EA3

pe5sF9-1H4-p2#comment-2759

- [x] Networking layer: Parse active features #10694
- [x] Update core data model: Store active features #10697
- [x] Add eligibility checker class #10700
- [ ] Use eligibility checker in coordinator #10710

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #10689

## Description

Checks for AI eligibility in coordinator. 

Changes
- Check the feature flag from the eligibility checker
- Receive store has products info in `isEligible` method instead of `init` method. This is needed to allow dependency injection and unit testing
- Fix the missed storage property update in https://github.com/woocommerce/woocommerce-ios/pull/10710/commits/c02c2e6195d88ab0c45727c87f815bb55d17bf84
- Add/update unit tests

## Testing instructions
- Create a JN site with Jetpack AI plugin. (Checkbox available in JN home page to install this plugin)
- Tap the add new product task on store onboarding. 
- Validate that the new AI action sheet is displayed.
- Switch to Products tab, select the "+" button to add a new product. The same AI action sheet should be displayed.
- Switch to a JN site with no Jetpack AI plugin and validate that you don't see the AI action sheet.

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift [hunk 1]
```diff
@@ -98,8 +103,7 @@ final class AddProductCoordinator: Coordinator {
         ServiceLocator.analytics.track(event: .ProductCreation.addProductStarted(source: source,
                                                                                  storeHasProducts: storeHasProducts))
 
-        // TODO-10688: Replace this with eligibility check
-        let isEligibleForAI = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.productCreationAI)
+        let isEligibleForAI = addProductWithAIEligibilityChecker.isEligible(storeHasProducts: storeHasProducts)
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift:106]: 👀 We should also update `storeHasProduct` to filter out sample products for free trial stores [pecCkj-Zo-p2#comment-506]

**Review by itsmeichigo — APPROVED** [review]
LGTM 🚢 

Nit: For `ProductCreationAIEligibilityCheckerTests` there should be a test for a non-WPCom store with AI assistant feature active.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift:106]: This ☝️ requires updating Networking and Yosemite, I'll take care of that.

**Comment by selanthiraiyan** [comment]
> For ProductCreationAIEligibilityCheckerTests there should be a test for a non-WPCom store with AI assistant feature active.

Done in https://github.com/woocommerce/woocommerce-ios/pull/10710/commits/917c7d955fa40942440952b8907ddcc4055e9158

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift:106]: Thank you! 🙇

**Pull request merged**